*** aoflagger/CMakeLists.txt	2018-10-19 11:27:20.671541615 +0200
--- CMakeLists.txt	2018-10-19 11:27:40.277968536 +0200
***************
*** 8,15 ****
  set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/CMake)
  
  find_package(PkgConfig)
! pkg_check_modules(GTKMM gtkmm-3.0>=3.0.0)
! pkg_check_modules(SIGCXX sigc++-2.0)
  
  SET(CASACORE_MAKE_REQUIRED_EXTERNALS_OPTIONAL TRUE)
  find_package(Casacore REQUIRED COMPONENTS casa ms tables measures)
--- 8,15 ----
  set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/CMake)
  
  find_package(PkgConfig)
! #pkg_check_modules(GTKMM gtkmm-3.0>=3.0.0)
! #pkg_check_modules(SIGCXX sigc++-2.0)
  
  SET(CASACORE_MAKE_REQUIRED_EXTERNALS_OPTIONAL TRUE)
  find_package(Casacore REQUIRED COMPONENTS casa ms tables measures)
