--- AOFlaggerStep.cc    2023-10-11 19:59:31.611562281 +0200
+++ AOFlaggerStep_fix.cc        2023-10-11 20:00:25.814211355 +0200
@@ -48,8 +48,8 @@
         aoflagger_.FindStrategyFile(aoflagger::TelescopeId::LOFAR_TELESCOPE);

   window_size_ = parset.getUint(prefix + "timewindow", 0);
-  memory_ = parset.getUint(prefix + "memorymax", 0);
-  memory_percentage_ = parset.getUint(prefix + "memoryperc", 0);
+  memory_ = parset.getDouble(prefix + "memorymax", 0);
+  memory_percentage_ = parset.getDouble(prefix + "memoryperc", 0.0);
   overlap_ = parset.getUint(prefix + "overlapmax", 0);
   // Also look for keyword overlap for backward compatibility.
   if (overlap_ == 0) {