*** Makefile_array	2018-10-23 16:48:59.134985812 +0200
--- killMS/Array/Dot/Makefile	2018-10-23 16:24:30.645281807 +0200
***************
*** 1,3 ****
--- 1,5 ----
+ NUMPY_LOCATION = $(shell python -c "import numpy; NUM_DIR=numpy.__file__.split('__init__.py')[0]; print(NUM_DIR+'core/include/numpy')")
+ PYTHON_LIBRARY_LOCATION =/usr/include/python2.7
  all: dotSSE.so
  
  # remove compilation products
***************
*** 11,14 ****
  
  # ---- gcc C compile ------------------
  dotSSE.o:  dotSSE.c dotSSE.h
! 	gcc -O3 --fast-math  -c dotSSE.c -I/usr/include/python2.7 -I/home/cyril.tasse/bin/lib/python2.7/site-packages/numpy/core/include/numpy  -I/home/tasse/bin/lib/python2.7/site-packages/numpy/core/include/numpy -I/home/tasse/bin/lib64/python2.7/site-packages/numpy-1.8.0-py2.7-linux-x86_64.egg/numpy/core/include/numpy -I/usr/lib/python2.7/dist-packages/numpy/core/include/numpy -I/home/cyril.tasse/bin/lib/python2.7/site-packages/numpy/core/include/numpy -I/home/tasse/bin/lib64/python2.7/site-packages/numpy-1.8.0-py2.7-linux-x86_64.egg/numpy/core/include/numpy -fPIC
--- 13,16 ----
  
  # ---- gcc C compile ------------------
  dotSSE.o:  dotSSE.c dotSSE.h
! 	gcc -O3 --fast-math  -c dotSSE.c -I${PYTHON_LIBRARY_LOCATION} -I${NUMPY_LOCATION} -fPIC
