--- jobserver.py	2021-07-16 14:10:57.171706750 +0200
+++ jobserver_new.py	2021-07-16 14:11:19.286011246 +0200
@@ -110,7 +110,7 @@
     ):
         if not host:
             host = socket.gethostname()
-        SocketServer.ThreadingTCPServer.__init__(self, (host, port), JobStreamHandler)
+        SocketServer.ThreadingTCPServer.__init__(self, ('0.0.0.0', port), JobStreamHandler)
         self.abort = False
         self.timeout = 1
         self.queue = Queue.Queue()
