*** Makefile	2019-01-10 16:02:01.641975817 +0100
--- Makefile_new	2019-01-10 16:03:26.374415632 +0100
***************
*** 2,12 ****
  _pyGridder.so:  Gridder.o 
  	gcc -fopenmp -pthread -shared Gridder.o -o _pyGridder.so
  
  # ---- gcc C compile ------------------
  Gridder.o:  Gridder.c Gridder.h
  #	gcc -O3 -msse -msse2 -msse2avx -msse3 -msse4.2 -mssse3 -march=corei7 -mtune=corei7 -c Gridder.c -I/usr/include/python2.7 -I/usr/lib/python2.7/dist-packages/numpy/core/include/numpy -fPIC	
  
! 	gcc  -lgomp -fopenmp -pthread -Ofast --fast-math  -c Gridder.c -I/usr/include/python2.7 -I/usr/lib/python2.7/dist-packages/numpy/core/include/numpy -I/home/cyril.tasse/bin/lib/python2.7/site-packages/numpy/core/include/numpy -I/home/tasse/bin/lib64/python2.7/site-packages/numpy-1.8.0-py2.7-linux-x86_64.egg/numpy/core/include/numpy -fPIC	
  #	gcc  -lgomp -fopenmp -pthread -O3 -msse -msse2 -msse3 -msse4.2 -mssse3 -march=corei7 -mtune=corei7 --fast-math  -c Gridder.c -I/usr/include/python2.7 -I/usr/lib/python2.7/dist-packages/numpy/core/include/numpy -fPIC	
  
  #gcc -c Gridder.c -I/usr/include/python2.7 -I/usr/lib/python2.7/dist-packages/numpy/core/include/numpy -fPIC
--- 2,15 ----
  _pyGridder.so:  Gridder.o 
  	gcc -fopenmp -pthread -shared Gridder.o -o _pyGridder.so
  
+ NUMPY_LOCATION = $(shell python -c "import numpy; NUM_DIR=numpy.__file__.split('__init__.py')[0]; print(NUM_DIR+'core/include/numpy')")
+ PYTHON_LIBRARY_LOCATION =/usr/include/python2.7
+ 
  # ---- gcc C compile ------------------
  Gridder.o:  Gridder.c Gridder.h
  #	gcc -O3 -msse -msse2 -msse2avx -msse3 -msse4.2 -mssse3 -march=corei7 -mtune=corei7 -c Gridder.c -I/usr/include/python2.7 -I/usr/lib/python2.7/dist-packages/numpy/core/include/numpy -fPIC	
  
! 	gcc  -lgomp -fopenmp -pthread -Ofast --fast-math  -c Gridder.c -I/usr/include/python2.7 -I${NUMPY_LOCATION} -fPIC	
  #	gcc  -lgomp -fopenmp -pthread -O3 -msse -msse2 -msse3 -msse4.2 -mssse3 -march=corei7 -mtune=corei7 --fast-math  -c Gridder.c -I/usr/include/python2.7 -I/usr/lib/python2.7/dist-packages/numpy/core/include/numpy -fPIC	
  
  #gcc -c Gridder.c -I/usr/include/python2.7 -I/usr/lib/python2.7/dist-packages/numpy/core/include/numpy -fPIC
