*** 2018_10_10/lofar/src/CMake/variants/GNUCXX11.cmake	2018-10-11 17:08:11.182529257 +0200
--- 2018_16_10_test2/lofar/src/CMake/variants/GNUCXX11.cmake	2018-10-17 12:06:04.513822903 +0200
***************
*** 11,20 ****
  
  # GNU compiler suite
  set(GNUCXX11_COMPILERS GNUCXX11_C GNUCXX11_CXX GNUCXX11_Fortran GNUCXX11_ASM)
! set(GNUCXX11_C         /usr/bin/gcc)      # GNU C compiler
! set(GNUCXX11_CXX       /usr/bin/g++)      # GNU C++ compiler
! set(GNUCXX11_Fortran   /usr/bin/gfortran) # GNU Fortran compiler
! set(GNUCXX11_ASM       /usr/bin/gcc)      # GNU assembler
  
  set(GNUCXX11_C_FLAGS          "-W -Wall -Wno-unknown-pragmas")
  set(GNUCXX11_C_FLAGS_DEBUG    "-g")
--- 11,20 ----
  
  # GNU compiler suite
  set(GNUCXX11_COMPILERS GNUCXX11_C GNUCXX11_CXX GNUCXX11_Fortran GNUCXX11_ASM)
! #set(GNUCXX11_C         /usr/bin/gcc)      # GNU C compiler
! #set(GNUCXX11_CXX       /usr/bin/g++)      # GNU C++ compiler
! #set(GNUCXX11_Fortran   /usr/bin/gfortran) # GNU Fortran compiler
! #set(GNUCXX11_ASM       /usr/bin/gcc)      # GNU assembler
  
  set(GNUCXX11_C_FLAGS          "-W -Wall -Wno-unknown-pragmas")
  set(GNUCXX11_C_FLAGS_DEBUG    "-g")
